src/arrayMethodSort.js

```diff
@@ -4,8 +4,43 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const arrayCopy = [...this];
+
+    const defaultCompare = (a, b) => {
+      const aStr = String(a);
+      const bStr = String(b);
+
+      if (aStr < bStr) {
+        return -1;
+      }
+
+      if (aStr > bStr) {
+        return 1;
+      }
+
+      return 0;
+    };
+
+    const comparator =
+      typeof compareFunction === 'function' ? compareFunction : defaultCompare;
```

Comment:
The comparator is correctly assigned based on whether a compareFunction is provided. This is a good use of the ternary operator.